pipelines/release-pipeline.yml

```diff
@@ -1,59 +1,71 @@
 trigger: none
 
 parameters:
-- name: linuxImage
+- name: buildAgentPoolVar
+  displayName: 'Build agent pool'
   type: string
-  default: 'ubuntu-18.04'
-  values:
-  - 'ubuntu-18.04'
-  - 'ubuntu-latest'
-- name: windowsImage
+  default: 'BuildAgentPool'
+- name: windowsBuildAgentVmImageVar
+  displayName: 'Windows build agent image'
   type: string
-  default: 'windows-2019'
-  values:
-  - 'windows-2019'
-  - 'windows-latest'
-- name: macImage
+  default: 'WindowsBuildAgentImage'
+- name: linuxBuildAgentVmImageVar
+  displayName: 'Linux build agent image'
   type: string
-  default: 'macOS-10.15'
-  values:
-  - 'macOS-10.15'
-  - 'macOS-latest'
+  default: 'LinuxBuildAgentImage'
+- name: release
+  displayName: 'Publish Release'
+  type: boolean
+  default: false
+
+variables:
+  - name: winVmImage
+    value: $[variables.${{ parameters.windowsBuildAgentVmImageVar }}]
+  - name: linuxVmImage
+    value: $[variables.${{ parameters.linuxBuildAgentVmImageVar }}]
 
 stages:
-  - stage: 'build'
+  - stage: 'SDL'
+    displayName: 'SDL Stage'
+    jobs:
+    - job: SDL_checks
+      displayName: 'SDL checks'
+      pool:
+        vmImage: windows-latest
+      steps:
+        - template: 'common/sdl-checks.yml'
+  - stage: 'Test'
     displayName: 'Build and Test'
+    dependsOn: [SDL]
     pool:
-      vmImage: ${{ parameters.linuxImage }}
+      vmImage: ubuntu-latest
```

Comment:
would it be helpful to have this as a variable so the user can set what vm image to use during testing rather than defaulting to latest? (similar to what we have in the azure iot extension)

Also applies to the other images/versions

Reply:
While this is a release pipeline - it does beg the question of what happens when our builds may suddenly not succeed if a `-latest` gets bumped to a new agent OS.

I'll add some parameters to the pipeline so that it can be run with various non-default agent images in the case that we need a build to succeed without debugging issues in a new OS.